istioctl/cmd/waypoint.go

```diff
@@ -0,0 +1,139 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	gateway "sigs.k8s.io/gateway-api/apis/v1beta1"
+	"sigs.k8s.io/yaml"
+
+	"istio.io/istio/istioctl/pkg/util/handlers"
+	"istio.io/istio/pkg/config/constants"
+	"istio.io/istio/pkg/config/schema/gvk"
+)
+
+func waypointCmd() *cobra.Command {
+	var waypointServiceAccount string
+	makeGateway := func(forApply bool) *gateway.Gateway {
+		ns := handlers.HandleNamespace(namespace, defaultNamespace)
+		if namespace == "" && !forApply {
+			ns = ""
+		}
+		name := waypointServiceAccount
+		if name == "" {
+			name = "namespace"
+		}
+		gw := gateway.Gateway{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       gvk.KubernetesGateway.Kind,
+				APIVersion: gvk.KubernetesGateway.GroupVersion(),
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
```

Comment:
This is missing the `--waypoint` suffix, which is in https://github.com/istio/istio/issues/43279 issue's template.

Reply:
Issue is wrong, the Deployment name gets `-istio-waypoint` added already so putting `waypoint` twice is redundant